docs/ug/authorConfigSyntax.md

```diff
@@ -0,0 +1,107 @@
+|-------------------------------------------------|---------|----------------------|-----------------------|
+| https://github.com/reposense/RepoSense.git      | master  | sikai00              | --                    |
+| https://github.com/reposense/RepoSense.git      | cypress | sikai00              | --                    |
+| https://github.com/reposense/RepoSense.git      | release | sikai00              | --                    |
+| https://github.com/reposense/testRepo-Alpha.git | master  | sikai00              | --                    |
+
+In order to provide details for the author for all the repositories, we would have to duplicate each row containing the same author details, but for each individual repository!
+
+By specifying multiple repository locations per row, we can prevent duplication of the author's information and keep everything within a single row. 
+
+| Repository's Location                                                                                                | Branch  | Author's Git Host ID | ... Hidden columns    |
+|----------------------------------------------------------------------------------------------------------------------|---------|----------------------|-----------------------|
+| https://github.com/reposense/RepoSense.git~master\|cypress\|release;https://github.com/reposense/testRepo-Alpha.git  | master  | sikai00              | --                    |
+
+<br>
+
+### Usage
+
+#### Specifying multiple `Repository's Location`
+
+To specify multiple `Repository's Location`, we can use a semicolon `;` as a separator.
+
+Below is an example:
+
+| Repository's Location                                                               | Branch  | Author's Git Host ID | ... Hidden columns    |
+|-------------------------------------------------------------------------------------|---------|----------------------|-----------------------|
+| https://github.com/reposense/RepoSense.git;https://github.com/MarkBind/markbind.git | master  | sikai00              | --                    |
+
+We have now furnished author details for both repositories in a single row, instead of having two individual rows per Repository's Location.
+
+<br>
+
+#### Specifying multiple `Repository's Location` with a single specific branch
+
+To override the Branch column with a specific branch for a particular `Repository's Location`, we can use a tilde `~` separator to denote our branch.
+
+Below is an example:
+
+| Repository's Location                                                                       | Branch  | Author's Git Host ID | ... Hidden columns    |
+|---------------------------------------------------------------------------------------------|---------|----------------------|-----------------------|
+| https://github.com/reposense/RepoSense.git~release;https://github.com/MarkBind/markbind.git | master  | sikai00              | --                    |
```

Comment:
That's a good suggestion, I hadn't thought of doing it that way

Reply:
Actually, I believe I had a concern while doing this: It becomes difficult to tell if `|` is part of the actual file path, or whether it is being used as a delimiter.

If the author happens to have `|` as part of their file path, but also wants to use the delimiter, it becomes impossible to tell which is the branch and which is the actual file path.

My current implementation does this by only getting the very last delimiter using the particular syntax which I believe is more ideal, since the author is likely to use the `~` and `|` delibrately for the location.

Reply:
Ok I actually read the original post examples and didn't verify if the code worked like that but for some reason I had the impression that `~` was used only for remote repos (see example 4). Which is also why I was suggesting standardizing the remote vs local since I thought they were parsed separately as well.

Since that is the case, I do think it makes some sense to look for the last `~` and then check if the entire thing is a valid repo filepath, otherwise split at the last `~` since branch names can't have `~`. There is still technically one scenario that is indistinguishable (in your example) both repos existing
```
/User/sikai00/actual|filepath/RepoSense~master|cypress 
/User/sikai00/actual|filepath/RepoSense
```
but I think it's improbable enough to not be of concern. Overall though, I think we should still look to merge remote and local parsing into one.

Reply:
I see, in that case I suppose the examples can be better specified then. I will most likely be merging the remote and local parsing into one, as I do believe that will be more ideal.

> There is still technically one scenario that is indistinguishable (in your example) both repos existing

I am not sure on what you mean with this. Do you mean having both as Repository's Location as separate rows in `author-config.csv`?

Reply:
The user could in theory have a repository located in the coincidentally-named directory `/User/sikai00/actual|filepath/RepoSense~master|cypress` or he could have one in `/User/sikai00/actual|filepath/RepoSense` with branches `master` and `cypress` he wishes to analyze. 
If they specify one line of `/User/sikai00/actual|filepath/RepoSense~master|cypress` in the csv, I don't think it's possible to detect which one the user 'intended' though I don't think this is likely to occur.

Reply:
I see, that is indeed a really obscure and improbable case. I currently don't have any good ideas on how this might be resolved actually. An possible solution might be to add a new column just to denote that the syntax is being used, but this affects usability since the user will need to indicate in a new column which goes against the purpose of the PR. Do you think it'll be good to add a small note, or should I disregard it entirely? I am fine with either cases, though I do believe that the note is a bit of an overkill, but at the same time it doesn't sit right to leave it as is.

Reply:
I think it's not something we have to handle because it just seems too improbable. Perhaps you could add a comment near the parsing with the link to this thread detailing the possible loophole in parsing repo + branches in case it ever becomes an issue

Reply:
Completed in the most recent commit.